org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java

```diff
@@ -106,11 +107,14 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
 			params.setTextDocument(identifier);
 			params.setPosition(LSPEclipseUtils.toPosition(offset, document));
 
-
-			Optional<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> tmp
-				= LanguageServers.forDocument(document).withFilter(LSPRenameProcessor::isPrepareRenameProvider)
-				.computeFirst((w, ls) -> ls.getTextDocumentService().prepareRename(params).thenApply(result -> new Pair<>(w, result)))
-				.get(1000, TimeUnit.MILLISECONDS);
+			@SuppressWarnings("null")
+			Optional<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> tmp =
+					LanguageServers.forDocument(document).withFilter(LSPRenameProcessor::isPrepareRenameProvider)
+					.collectAll((w, ls) -> ls.getTextDocumentService().prepareRename(params)
+							.thenApply(result -> new Pair<>(w, result)))
+					.get(1000, TimeUnit.MILLISECONDS).stream()
+					.filter(Objects::nonNull).filter(t-> t.getSecond() != null)

```

Comment:
This filter is actually invalid: there are "good" LS that would return a Range or a PrepareRenameDefaultBehavior and for which `getSecond()` would be null, and now filtered out here, while the `rename` request can work.

Reply:
Lost the answer... So, briefly repeating here...

Actually it's not. Whatever is returned by a LS is stored in `tmp.getSecond()` so, independently of  type of the returned object, the rename operation will be continued if it's not `null`.